[Tabs] Deduplicate item view layout and contentFrame calculation.

romoore opened the pull request:
Combines the code that performs layout and content frame calculation.
    
Part of #7748

material-automation commented:
bazel detected changes to the following targets:

```
//components/Tabs:ObjcExamples
//components/Tabs:TabBarView
//components/Tabs:snapshot_test_lib
//components/Tabs:snapshot_tests
//components/Tabs:snapshot_tests.swift_runtime_linkopts
//components/Tabs:snapshot_tests_test_binary
//components/Tabs:snapshot_tests_test_bundle
//components/Tabs:unit_test_tabbarview_sources
//components/Tabs:unit_tests
//components/Tabs:unit_tests_IPAD_PRO_12_9_IN_9_3
//components/Tabs:unit_tests_IPAD_PRO_12_9_IN_9_3.swift_runtime_linkopts
//components/Tabs:unit_tests_IPAD_PRO_12_9_IN_9_3_test_binary
//components/Tabs:unit_tests_IPAD_PRO_12_9_IN_9_3_test_bundle
//components/Tabs:unit_tests_IPHONE_7_PLUS_IN_10_3
//components/Tabs:unit_tests_IPHONE_7_PLUS_IN_10_3.swift_runtime_linkopts
//components/Tabs:unit_tests_IPHONE_7_PLUS_IN_10_3_test_binary
//components/Tabs:unit_tests_IPHONE_7_PLUS_IN_10_3_test_bundle
//components/Tabs:unit_tests_IPHONE_X_IN_11_0
//components/Tabs:unit_tests_IPHONE_X_IN_11_0.swift_runtime_linkopts
//components/Tabs:unit_tests_IPHONE_X_IN_11_0_test_binary
//components/Tabs:unit_tests_IPHONE_X_IN_11_0_test_bundle
```

<!-- identifier: affected-targets -->

Commit:
[Tabs] Improve text and image rendering.

The frames for title and image were not pixel-aligned, resulting in somewhat
blurry text and images.

Part of #7748

Commit:
[Tabs] Simplify item view layout and contentFrame calculation.

Combines the code that performs layout and content frame calculation.

Part of #7748

Commit:
Standardize by using `setFrame` since the two rounding methods were not
equivalent.

Commit:
Fix layout where label attempted to be too tall.

Commit:
Merge branch 'develop' into issue-7748-c
